addons/pos_self_order/controllers/utils.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+
+import werkzeug
+
+from odoo.http import request
+from odoo.addons.http_routing.models.ir_http import unslug
+from odoo.addons.pos_self_order.models.pos_config import PosConfig
+from odoo.addons.pos_restaurant.models.pos_restaurant import RestaurantTable
+
+
+def get_pos_config_sudo(pos_config_name: str) -> PosConfig:
+    """
+    Returns the PosConfig if pos_config_id exist and the pos is configured to allow the menu to be viewed online.
+    If not, it raises a NotFound
+    :param pos_config_name: The name of the pos config. Can be the id or the slug. ex: 3 or Bar-3
+    """
+    return (
+        request.env["pos.config"]
+        .sudo()
+        .search(
+            [
+                ("id", "=", unslug(str(pos_config_name))[1]),
+                ("self_order_view_mode", "=", True),
+            ],
+            limit=1,
+        )
+    ) or _raise(werkzeug.exceptions.NotFound)
```

Comment:
In some places, the error is instantiated, but here, it's not. Which is correct?